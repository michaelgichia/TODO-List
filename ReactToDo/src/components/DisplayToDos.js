import React, { Component } from 'react';
import _ from 'lodash';

export default class DisplayToDos extends Component {
	constructor(props){
    super(props)
    this.state = {
    	isEditing: false
    }
    this.editTodo = this.editTodo.bind(this)
    this.formSubmit = this.formSubmit.bind(this)
  }

  editTodo() {
  	this.setState({isEditing: true})
  }

  formSubmit(event) {
  	event.preventDefault()
  	const newContent = this.refs.newValue.value
  	const oldContent = this.props.content
  	this.props.updateTodos(newContent, oldContent)
  	this.setState({isEditing: false})
  }

   render() {
   	if(this.state.isEditing) {
   		return(
   			<form onSubmit={this.formSubmit}>
   				<input
   					type="text"
   					defaultValue={this.props.content}
   					ref="newValue"
   				/>
   			</form>
   		)
   	}
		return(
			<li key={this.props.keys}>
				<div className="todo-row">
					<input type="checkbox"/>
					<label>{this.props.content}</label>
					<button onClick={this.editTodo}>Edit</button>
					<button onClick={this.props.onDelete}>Delete</button>
				</div>
			</li>
		)
  }
}
 
