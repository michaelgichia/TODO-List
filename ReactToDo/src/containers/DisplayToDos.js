import React, {Component, PropTypes} from 'react';

export default class DisplayToDos extends Component {
	constructor(props){
    super(props)
    this.state = {
    	isEditing: false
    }
  }

  editTodo = () => {
  	this.setState({isEditing: true})
  }

  formSubmit = (event) => {
    event.preventDefault()
    const newContent = this.refs.newValue.value
    const oldContent = this.props.content
    this.props.updateTodos(newContent, oldContent)
    this.setState({isEditing: false})
  }

  toggle = () => {
    const index = this.refs.checkbox.id;
    const oldCheckboxVal = this.props.todos[index];
    this.props.updateCheckbox(oldCheckboxVal)
  }

  render() {
   	if(this.state.isEditing) {
   		return (
   			<form onSubmit={this.formSubmit}>
          <label>editing...</label>
          <input
            className="input-style"
            type="text"
            defaultValue={this.props.content}
            ref="newValue"
          />
   			</form>
   		)
   	}
		return (
			<li key={this.props.keys} className="z-depth-2">
        <input
          type="checkbox"
          className="filled-in"
          id={this.props.id}
          ref="checkbox"
          checked={this.props.isCompleted ? "checked":""}
          onChange={this.toggle}
        />
        <label htmlFor={this.props.id}>{this.props.content}</label>
        <div className="actions">
          <label
            className="edit transparent"
            onClick={this.editTodo}>
            <i className="tiny material-icons">mode_edit</i>
          </label>
          <label
          className="delete transparent"
          onClick={this.props.onDelete}>
            &times;
          </label>
        </div>
      </li>
    )
  }
}
 
DisplayToDos.propTypes = {
  updateTodos: PropTypes.func,
  onDelete: PropTypes.func,
  content: PropTypes.string,
  todos: PropTypes.array,
  updateCheckbox: PropTypes.func,
  keys: PropTypes.number,
  id: PropTypes.number
}