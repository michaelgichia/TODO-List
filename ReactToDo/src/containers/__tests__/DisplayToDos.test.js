import React from 'react'
import { spy } from 'sinon';
import { mount, shallow } from 'enzyme';

import DisplayToDos from '../DisplayToDos'

describe('App components', () => {
  const onDeleteClick = spy();
  const props = {
    todos: [
      {content: "Learn Redux in Depth", isCompleted: true}
      ],
    onDelete: onDeleteClick(),
    id: 0,
    htmlFor: 0

  };
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <DisplayToDos {...props}/>
    );
  });

  it('renders correctly', () => {
    const tree = shallow(
      <DisplayToDos />
    );
    expect(tree).toMatchSnapshot();
  });

  it('delete todo when delete clicked', () => {
    wrapper.find('.delete').simulate('click');
    expect(onDeleteClick.calledOnce).toBe(true);
  });

})


