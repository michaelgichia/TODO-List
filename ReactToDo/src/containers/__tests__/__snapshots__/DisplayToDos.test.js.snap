// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <li
    className="z-depth-2"
>
    <input
        checked=""
        className="filled-in"
        id={undefined}
        onChange={[Function]}
        type="checkbox"
    />
    <label
        htmlFor={undefined}
    />
    <div
        className="actions"
    >
        <label
            className="edit transparent"
            onClick={[Function]}
        >
            <i
                className="tiny material-icons"
            >
                mode_edit
            </i>
        </label>
        <label
            className="delete transparent"
            onClick={undefined}
        >
            ×
        </label>
    </div>
</li>,
  "nodes": Array [
    <li
      className="z-depth-2"
>
      <input
            checked=""
            className="filled-in"
            id={undefined}
            onChange={[Function]}
            type="checkbox"
      />
      <label
            htmlFor={undefined}
      />
      <div
            className="actions"
      >
            <label
                  className="edit transparent"
                  onClick={[Function]}
            >
                  <i
                        className="tiny material-icons"
                  >
                        mode_edit
                  </i>
            </label>
            <label
                  className="delete transparent"
                  onClick={undefined}
            >
                  ×
            </label>
      </div>
</li>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DisplayToDos />,
      "_debugID": 10,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DisplayToDos {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "editTodo": [Function],
        "formSubmit": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "isEditing": false,
        },
        "toggle": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <li
          className="z-depth-2"
>
          <input
                    checked=""
                    className="filled-in"
                    id={undefined}
                    onChange={[Function]}
                    type="checkbox"
          />
          <label
                    htmlFor={undefined}
          />
          <div
                    className="actions"
          >
                    <label
                              className="edit transparent"
                              onClick={[Function]}
                    >
                              <i
                                        className="tiny material-icons"
                              >
                                        mode_edit
                              </i>
                    </label>
                    <label
                              className="delete transparent"
                              onClick={undefined}
                    >
                              ×
                    </label>
          </div>
</li>,
        "_debugID": 11,
        "_renderedOutput": <li
          className="z-depth-2"
>
          <input
                    checked=""
                    className="filled-in"
                    id={undefined}
                    onChange={[Function]}
                    type="checkbox"
          />
          <label
                    htmlFor={undefined}
          />
          <div
                    className="actions"
          >
                    <label
                              className="edit transparent"
                              onClick={[Function]}
                    >
                              <i
                                        className="tiny material-icons"
                              >
                                        mode_edit
                              </i>
                    </label>
                    <label
                              className="delete transparent"
                              onClick={undefined}
                    >
                              ×
                    </label>
          </div>
</li>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DisplayToDos />,
}
`;
